disdrodb/data_transfer/download_data.py

```diff
@@ -0,0 +1,282 @@
+        url=url,
+        known_hash=None,
+        path=folder_path,
+        fname=file_name,
+        downloader=downloader,
+        progressbar=tqdm,
+    )
+
+
+# function to unzip one file into a folder
+def unzip_file(file_path: str, dest_path: str) -> None:
+    """Unzip a file into a folder
+
+    Parameters
+    ----------
+    file_path : str
+        Path of the file to unzip
+    dest_path : str
+        Path of the destination folder
+    """
+
+    with zipfile.ZipFile(file_path, "r") as zip_ref:
+        zip_ref.extractall(dest_path)
+
+
+def download_file_content(url_local_path: tuple, overwrite: bool = False, unzip: bool = False) -> None:
+    """Download the files based on a tuple (local path , url).
+    Unzip the file if it is a zip file and if requested.
+
+    Parameters
+    ----------
+    file_path : tuple
+        Tuple containing the local file path and the url
+    overwrite : bool, optional
+        Overwrite the raw data file is already existing, by default False
+    unzip : bool, optional
+        Unzip the file if it is a zip file, by default False
+    """
+
+    local_path, url = url_local_path
+    if check_url(url):
+        # create station folder if not existing
+        if not os.path.exists(local_path):
+            os.makedirs(local_path)
+        # download the file
+        download_file_from_url(url, local_path)
+        print(f"Download {url} into {local_path}")
+        if url.endswith(".zip") and unzip:
+            zip_local_path = os.path.join(local_path, os.path.basename(url))
+            unzip_file(zip_local_path, local_path)
+            os.remove(zip_local_path)
+
+    else:
+        # raise an error if url not well formated
+        raise ValueError(f"URL {url} is not well formated. Please check the URL.")
+
+
+def download_all_files(
+    raw_folder: str = None,
+    data_source: str = None,
+    campaign_name: str = None,
+    station_name: str = None,
+    overwrite: bool = False,
+    unzip: bool = False,
+):
+    """Batch function to get all yaml files that coantain
+    the 'data_url' key and download the data locally.
+
+    Parameters
+    ----------
+    raw_folder : str, optional
+        Raw data folder path.
+        Must end with DISDRODB/Raw
+    data_source : str, optional
+        Data source folder name (eg : EPFL).
+        If not provided (None), all data sources will be downloaded.
+        The default is data_source=None
+    campaign_name : str, optional
+        Campaign name (eg :  EPFL_ROOF_2012).
+        If not provided (None), all campaigns will be downloaded.
+        The default is campaign_name=None
+    station_name : str, optional
+        Station name.
+        If not provided (None), all stations will be downloaded.
+        The default is station_name=None
+    overwrite : bool, optional
+        If True, overwrite the already existing raw data file.
+        The default is False.
+    unzip : bool, optional
+        If True, unzip the zip file.

```

Comment:
To clarify behavior. 
I expect to upload on Zenodo a zip file of the stations data directory (or multiple zipped directories if for example the station data directory is something like data/<station_name>/<year>/... 
I would not unzip the files contained in the directories ... just a waste of disk storage and would make the reader not working because the `file_pattern` would change ! 

Would be nice to have a utility that enable to automatically zip/gzip/whatever compress each file ... so that data contributors can easy compress their files, then adapt the reader file_pattern and then upload the data

Reply:
no zip anymore